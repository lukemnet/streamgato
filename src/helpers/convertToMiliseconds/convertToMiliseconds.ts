export interface CountdownTimerParams {
  h?: number;
  m?: number;
  s: number;
}

const convertToMiliseconds = ({ h, m, s }: CountdownTimerParams) => {
  const hMs = h ? h * 60 * 60 * 1000 : 0;
  const mMs = m ? m * 60 * 1000 : 0;
  const sMs = s ? s * 1000 : 0;
  return hMs + mMs + sMs + 0.9;
}

export default convertToMiliseconds;
