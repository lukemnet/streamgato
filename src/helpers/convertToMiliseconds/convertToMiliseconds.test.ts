import convertToMiliseconds from './convertToMiliseconds';

const testValues = [
  {
    s: 5
  },
  {
    m: 15,
    s: 45,
  },
  {
    h: 4,
    m: 55,
    s: 22,
  },
  {
    h: 48,
    m: 65,
    s: 56,
  },
  {
    h: 22,
    s: 43,
  },
  {
    h: 9,
    m: 49,
  },
];

it('should convert numbers to miliseconds', () => {
  testValues.forEach(value => {
    expect(convertToMiliseconds(value)).toMatchSnapshot();
  });
});
