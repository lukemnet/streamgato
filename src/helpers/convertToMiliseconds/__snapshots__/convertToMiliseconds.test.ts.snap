// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert numbers to miliseconds 1`] = `5000.9`;

exports[`should convert numbers to miliseconds 2`] = `945000.9`;

exports[`should convert numbers to miliseconds 3`] = `17722000.9`;

exports[`should convert numbers to miliseconds 4`] = `176756000.9`;

exports[`should convert numbers to miliseconds 5`] = `79243000.9`;

exports[`should convert numbers to miliseconds 6`] = `35340000.9`;
